arch/xtensa/src/esp32s3/esp32s3_spiflash_mtd.c

```diff
@@ -235,6 +235,322 @@ static inline bool IRAM_ATTR stack_is_psram(void)
 }
 #endif
 
+/**
+ * From: esp-idf/components/hal/esp32s3/include/hal/spi_flash_encrypted_ll.h
+ */
+
+/**
+ * Choose type of chip you want to encrypt manully
+ */
+
+typedef enum
+{
+    FLASH_ENCRYPTION_MANU = 0, /* !< Manually encrypt the flash chip. */
+    PSRAM_ENCRYPTION_MANU = 1  /* !< Manually encrypt the psram chip. */
+} flash_encrypt_ll_type_t;
+
+/**
+ * Enable the flash encryption function under spi boot mode and download boot
+ * mode.
+ */
+
+static inline void spi_flash_encrypt_ll_enable(void)
+{
+    REG_SET_BIT(SYSTEM_EXTERNAL_DEVICE_ENCRYPT_DECRYPT_CONTROL_REG,
+                SYSTEM_ENABLE_DOWNLOAD_MANUAL_ENCRYPT |
+                SYSTEM_ENABLE_SPI_MANUAL_ENCRYPT);
+}
+
+/**
+ * Disable the flash encryption mode.
+ */
+
+static inline void spi_flash_encrypt_ll_disable(void)
+{
+    REG_CLR_BIT(SYSTEM_EXTERNAL_DEVICE_ENCRYPT_DECRYPT_CONTROL_REG,
+                SYSTEM_ENABLE_SPI_MANUAL_ENCRYPT);
+}
+
+/**
+ * Choose type of chip you want to encrypt manully
+ *
+ * @param type The type of chip to be encrypted
+ *
+ * @note The hardware currently support flash encryption.
+ */
+
+static inline void spi_flash_encrypt_ll_type(flash_encrypt_ll_type_t type)
+{
+    HAL_ASSERT(type == FLASH_ENCRYPTION_MANU); /* Our hardware only support flash encryption */
```

Comment:
This macro shouldn't be used in NuttX. Use DEBUGASSERT, instead.:
```suggestion
    DEBUGASSERT(type == FLASH_ENCRYPTION_MANU); /* Our hardware only support flash encryption */
```

Reply:
Please check other occurrences.